src/Core/Animation/Skinning/RotationCenterSkinning.cpp

```diff
@@ -55,116 +51,161 @@ void computeCoR( Skinning::RefData& dataInOut, Scalar sigma, Scalar weightEpsilo
     LOG( logDEBUG ) << "Precomputing CoRs";
 
     // First step : subdivide the original mesh until weights are sufficiently close enough.
-    Scalar maxWeightDistance = 0.f;
-    TriangleMesh subdividedMesh;
-    subdividedMesh.copyBaseGeometry( dataInOut.m_referenceMesh );
+    // Convert the mesh to TopologicalMesh for easy processing.
+    TriangleMesh triMesh;
+    triMesh.copyBaseGeometry( dataInOut.m_referenceMesh );
+    TopologicalMesh topoMesh( triMesh );
+
+    // hashing function for Vector3
+    struct hash_vec {
+        size_t operator()( const Ra::Core::Vector3& lvalue ) const {
+            size_t hx = std::hash<Scalar>()( lvalue[0] );
+            size_t hy = std::hash<Scalar>()( lvalue[1] );
+            size_t hz = std::hash<Scalar>()( lvalue[2] );
+            return ( hx ^ ( hy << 1 ) ) ^ hz;
+        }
+    };
 
-    // Store the weights as row major here because we are going to query the per-vertex weights.
-    Eigen::SparseMatrix<Scalar, Eigen::RowMajor> subdividedWeights = dataInOut.m_weights;
+    // fill map from vertex position to handle index, used to access the weight
+    // matrix from initial mesh vertices
+    std::unordered_map<Ra::Core::Vector3, int, hash_vec> mapV2I;
+    int i = 0;
+    for ( auto vit = topoMesh.vertices_begin(); vit != topoMesh.vertices_end(); ++vit, ++i )
+    {
+        mapV2I[topoMesh.point( *vit )] = vit->idx();
+    }
 
-    // Convert the mesh to DCEL for easy processing.
-    Dcel dcel;
-    convert( subdividedMesh, dcel );
+    // Squash weight matrix to fit TopologicalMesh (access through handle indices)
+    // Store the weights as row major here because we are going to query the per-vertex weights.
```

Comment:
Change this to multiline comment.

Reply:
these are 2 different one-line comments, so what to do about it?